core/src/main/java/nl/nn/adapterframework/doc/model/FrankDocModel.java

```diff
@@ -1,45 +1,143 @@
+	
+	private @Getter Map<String, ConfigChildSetterDescriptor> configChildDescriptors;
 	private @Getter List<FrankDocGroup> groups;
 	private @Getter Map<String, FrankElement> allElements = new HashMap<>();
+	private @Getter Map<String, ElementType> allTypes = new HashMap<>();
+
+	/**
+	 * Get the FrankDocModel needed in production. This is just a first draft. The
+	 * present version does not have groups yet. It will be improved in future
+	 * pull requests. 
+	 */
+	public static FrankDocModel populate() {
+		FrankDocModel result = new FrankDocModel();
+		result.createConfigChildDescriptorsFrom(DIGESTER_RULES);
+		result.findOrCreateElementType(Utils.getClass("nl.nn.adapterframework.core.IAdapter"));
+		return result;
+	}
 
 	public FrankDocModel() {
+		configChildDescriptors = new HashMap<>();
 		groups = new ArrayList<>();
 	}
 
+	public void createConfigChildDescriptorsFrom(String path) {
+		Resource resource = Resource.getResource(path);
+		if(resource == null) {
+			throw new RuntimeException(String.format("Cannot find resource on the classpath: [%s]", path));
+		}
+		try {
+			XmlUtils.parseXml(resource.asInputSource(), new Handler(path));
+		}
+		catch(IOException e) {
+			throw new RuntimeException(String.format("An IOException occurred while parsing XML from [%s]", path), e);
+		}
+		catch(SAXException e) {
+			throw new RuntimeException(String.format("A SAXException occurred while parsing XML from [%s]", path), e);
+		}
+	}
+
+	private class Handler extends DigesterRulesHandler {
+		private final String path;
+
+		Handler(String path) {
+			this.path = path;
+		}
+
+		@Override
+		protected void handle(DigesterRule rule) throws SAXException {
+			String pattern = rule.getPattern();
+			StringTokenizer tokenizer = new StringTokenizer(pattern, "/");
+			String syntax1Name = null;
+			while(tokenizer.hasMoreElements()) {
+				String token = tokenizer.nextToken();
+				if(!"*".equals(token)) {
+					syntax1Name = token;
+				}
+			}
+			if(StringUtils.isNotEmpty(rule.getRegisterMethod())) {
+				add(rule.getRegisterMethod(), syntax1Name);
+			}			
+		}
+
+		private void add(String registerMethod, String syntax1Name) throws SAXException {
+			ConfigChildSetterDescriptor item = new ConfigChildSetterDescriptor(registerMethod, syntax1Name);
+			if(configChildDescriptors.containsKey(item.getMethodName())) {
+				log.warn(String.format("In digester rules [%s], duplicate method name [%s]", path, registerMethod));
+			} else {
+				configChildDescriptors.put(item.getMethodName(), item);
+			}
+		}
+	}
+
+	public ElementType findOrCreateElementType(Class<?> clazz) {
+		if(allTypes.containsKey(clazz.getName())) {
+			return allTypes.get(clazz.getName());
+		}
+		final ElementType result = new ElementType(clazz);
+		// If a containing FrankElement contains the type being created, we do not
+		// want recursion.
+		allTypes.put(result.getFullName(), result);
+		if(clazz.isInterface()) {
```

Comment:
Ik denk dat het ook handig is om de waarde van clazz.isInterface() te bewaren in ElementType result. Die gaat nog van pas komen denk ik.

Reply:
Gedaan.